src/facade/dragonfly_connection.cc

```diff
@@ -854,16 +854,19 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
       service_->DispatchManyCommands(absl::MakeSpan(args), cc_.get());
       cc_->async_dispatch = false;
 
+      // Flush strictly before the dispatch queue is cleared so that no sync dispatch can occur
+      if (dispatch_q_.size() == args.size())  // Flush if no new messages appeared
+        builder->FlushBatch();
+
+      DCHECK(!cc_->sync_dispatch);
+      builder->SetBatchMode(false);  // in case the next dispatch is sync
```

Comment:
But why disable batch mode unconditionally?

Reply:
It will be re-enabled either way in the next loop iteration if needed